src/planning/recipe-index.ts

```diff
@@ -184,7 +184,7 @@ export class RecipeIndex {
         if (connectedParticles.size === particleNames.length
             && particleNames.length === otherParticleNames.length) continue;
 
-        results.push(otherHandle);
+        results.push({handle: otherHandle, recipe});
```

Comment:
It could be worth actually defining a type for the return value, so that it's explicit where used elsewhere, what the fields/types are. Perhaps PlanningCandidate? PlanHandle? PlanHandleCandidate? Just a few thoughts.